examples/creatives/banner_nonlinear.js

```diff
@@ -0,0 +1,51 @@
+class BannerNonLinear extends BaseSimidCreative{
+    constructor() {
+        super();
+        this.addButtonClickActions_();
+    }
+
+    /** @override */
+    onInit(eventData) {
+        super.onInit(eventData);
+        this.dynamicResize_();
+    }
+
+    /**
+     * Adds actions to different buttons available on the overlay.
+     * @private
+     */
+    addButtonClickActions_() {
+        this.sendMessageOnButtonClick_("close_ad", CreativeMessage.REQUEST_STOP);
+        this.sendMessageOnButtonClick_("ad_text", CreativeMessage.REQUEST_EXPAND);
+        this.sendMessageOnButtonClick_("minimize_ad", CreativeMessage.REQUEST_COLLAPSE);
+    }
+
+    /**
+     * Listens for a click event on a button
+     * @param {String} elementName The name of the element.
+     * @param {String} message The message to send to the player.
+     * @private
+     */
+    sendMessageOnButtonClick_(elementName, message) {
+        const sendMessageFunction = () => {this.simidProtocol.sendMessage(message);}
+        document.getElementById(elementName).addEventListener(
+            'click', sendMessageFunction);
+    }
+    /**
+     * Repositions the banner ad according to the dimensions of the video player
+     * by calculating desired dimensions and sending a resize request to creative.
+     * @private
+     */
+    dynamicResize_() {
+        let creativeDimensions = {};
+        creativeDimensions.x = this.environmentData.videoDimensions.width * .15;
```

Comment:
You can add an explanation of why you are doing this here.

// This ad requests that the player resize it and move it, so that it is centered with 15% of the player size as border on the right and left.
// It will be 70% down from the top of the player.

Reply:
Added brief explanation